src/app/components/utility/userStateService.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Created by chima on 11/19/15.
+ */
+'use strict'
+
+angular.module('utility')
+  .service('userStateService', function (ehaCouchDbAuthService, locationService, $localForage, $q) {
+    var self = this
+    self.stateMap = {states: [], selectedState: ''}
+
+    function initializeStateVariables () {
+      $localForage.clear()
+      self.stateMap.states = []
+      self.stateMap.selectedState = ''
+    }
+
+    this.getUserSelectedState = function (byId) {
+      return ehaCouchDbAuthService.getCurrentUser()
+        .then(function (user) {
+          if (!user.userCtx.name) {
+            return ''
+          }
+          return $localForage.getItem(user.userCtx.name)
+            .then(function (state) {
+              var result = !byId ? state : (byId === true ? self.getState(state, true) : self.getState(state))
+              return $q.when(result)
+            })
+        })
+    }
+
+    this.setUserSelectedState = function (state) {
+      return ehaCouchDbAuthService.getCurrentUser()
+        .then(function (user) {
+          if (!state || !user.userCtx.name) {
+            return $q.when('')
+          }
+          return $localForage.setItem(user.userCtx.name, state)
+            .then(function (st) {
+              return true
+            })
+        })
+    }
+
+    this.getState = function (name, byId) {
+      return $localForage.getItem('states')
+        .then(function (states) {
+          if (!name || !states || !states.length) {
+            return ''
+          }
+          var state = states.filter(function (item) {
+            return item.name === name
+          })[0]
+          return byId ? state._id : state
+        })
+    }
+
+    this.getUserStates = function () {
+      function getStatesByUser (user) {
+        var LEVEL = '2'
+        if (user.isAdmin()) {
+          return locationService.getLocationsByLevel(LEVEL)
+        }
+        var stateIds = self.authorisedStates(user)
+        return locationService.getLocationsByLevelAndId(LEVEL, stateIds)
+      }
+
+      return ehaCouchDbAuthService.getCurrentUser().then(getStatesByUser)
+    }
+
+    this.authorisedStates = function (user) {
```

Comment:
Yea, their was cyclic dependency but i really needed that function, may be i would have to go with the option of implementing the userStatesService features inside AuthService